From ca8558b5e757c51dffac1322a721948473c0247f Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:39:02 -0500
Subject: [drm] nouveau/kms: restore acceleration before fb_set_suspend() resumes

Message-id: <1418852380-13061-52-git-send-email-rclark@redhat.com>
Patchwork-id: 102222
O-Subject: [RHEL7 drm 51/89] drm/nouveau/kms: restore acceleration before fb_set_suspend() resumes
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: 3.17

commit 447d2e44348f164ee9a465217c2b38c40cdb81ac

Author:     Ben Skeggs <bskeggs@redhat.com>
AuthorDate: Mon Jun 30 13:18:48 2014 +1000
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:45:24 2014 -0500

    drm/nouveau/kms: restore acceleration before fb_set_suspend() resumes

    commit 04384435fbcd2c5cb70b2e58c8b643822d7e337f upstream.

    This *should* be safe these days.

    Signed-off-by: Ben Skeggs <bskeggs@redhat.com>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/nouveau/nouveau_fbcon.c b/drivers/gpu/drm/nouveau/nouveau_fbcon.c
index 02dc2a1..2ef2842 100644
--- a/drivers/gpu/drm/nouveau/nouveau_fbcon.c
+++ b/drivers/gpu/drm/nouveau/nouveau_fbcon.c
@@ -550,13 +550,13 @@ nouveau_fbcon_set_suspend(struct drm_device *dev, int state)
  struct nouveau_drm *drm = nouveau_drm(dev);
  if (drm->fbcon) {
   console_lock();
-  if (state == 1)
-   nouveau_fbcon_accel_save_disable(dev);
-  fb_set_suspend(drm->fbcon->helper.fbdev, state);
   if (state == 0) {
    nouveau_fbcon_accel_restore(dev);
    nouveau_fbcon_zfill(dev, drm->fbcon);
   }
+  fb_set_suspend(drm->fbcon->helper.fbdev, state);
+  if (state == 1)
+   nouveau_fbcon_accel_save_disable(dev);
   console_unlock();
  }
 }
-- 
1.7.1